### DISCUSSION TITLE
Why does clicking ListBoxItems that have href not fire a click event

### DISCUSSION THREAD

**Discussion by fernandowinfield** [adobe/react-spectrum#6487] [Q&A]
Hello! My team and I have been using the React Aria Components library as the base for our common UI components; we really like it and appreciate all the effort put into it.

The question we have is: why does clicking on a ListBoxItem that has an `href` prop not fire a click event? Or is the click being suppressed for some reason? The reason why we are interested in this is because we have some event delegation set up (actually inspired by [this](https://gist.github.com/devongovett/919dc0f06585bd88af053562fd7c41b7)) to power some business logic.

In case it helps, I've put together [an example](https://codesandbox.io/p/sandbox/onclick-delegation-xqyxxf?file=%2Fsrc%2FApp.js%3A45%2C19) where the behavior mentioned above can be reproduced. Couple of notes:
- Our app uses Remix (I've set up the playground with React Router to keep it simple), so we use the RouterProvider from `react-aria-components` to enable client side routing.
  - Based on my testing, a click event **is** fired when passing a full URL as the `href`, but not when it is a relative path.
  - Click events are also captured if the React Router + RouterProvider setup is stripped out altogether.
- While `click` events aren't captured by the `onEvent` handler, `pointerup` events are, which is what we have opted to do in the meantime. The issue with using `pointerup` is that keyboard presses aren't captured that way.

**Comment by snowystinger** [comment]
We do a bunch of interaction normalization in our components https://react-spectrum.adobe.com/blog/building-a-button-part-1.html
And we do some extra work in collections to make links work across components. For instance, Table's also support links in the same manner, but they can't be `a` tags due to accessiblity https://react-spectrum.adobe.com/react-aria/Table.html#links.

As a result, if you're following that gist and we had left the click to fire, you'd get events for the ListBox, but not navigation for the Table.

We handle everything that gist would be handling for you in Collections, so it's better to follow the guides at https://react-spectrum.adobe.com/react-aria/routing.html#remix to set up your routing.

Let us know if that isn't fulfilling your needs.
